runtime/memory.c

```diff
@@ -729,6 +729,17 @@ CAMLexport caml_stat_string caml_stat_strdup(const char *s)
   return result;
 }
 
+CAMLexport caml_stat_string caml_stat_memdup(const char *s, asize_t size,
+                                             asize_t *out_size)
+{
+  CAMLassert(size > 0);
+  caml_stat_block result = caml_stat_alloc(size);
+  memcpy(result, s, size);
```

Comment:
This assumes `s` is at least `size` bytes long. How do you plan to guarantee this? Do you intend for all callers of `caml_stat_char_array_{of,to}_os` to enforce this?

Reply:
Yes - that’s the whole point of this function, processing buffers where the length is _specified_ not computed.